suite-native/transactions/src/components/TransactionsList/TransactionList.tsx

```diff
@@ -1,37 +1,83 @@
-import React from 'react';
-import { ActivityIndicator } from 'react-native';
+import React, { useRef } from 'react';
+import { ActivityIndicator, FlatList } from 'react-native';
+import { useSelector } from 'react-redux';
 
-import { A } from '@mobily/ts-belt';
+import { A, D } from '@mobily/ts-belt';
 
 import { useNativeStyles } from '@trezor/styles';
-import { WalletAccountTransaction } from '@suite-common/wallet-types';
+import { Account, WalletAccountTransaction } from '@suite-common/wallet-types';
 import { groupTransactionsByDate } from '@suite-common/wallet-utils';
+import { Box, Text } from '@suite-native/atoms';
+import { AccountBalance } from '@suite-native/accounts';
+import { AccountsRootState, selectAccountsTransactionsCount } from '@suite-common/wallet-core';
+import { TAB_BAR_HEIGHT } from '@suite-native/navigation';
 
 import { TransactionListItem } from './TransactionListItem';
 import { TransactionListGroup } from './TransactionListGroup';
+import { TransactionListPagination } from './TransactionListPagination';
 
 type AccountTransactionProps = {
     transactions: WalletAccountTransaction[];
+    accountName?: string;
+    account: Account;
+    page: number;
+    setPage: (page: number) => void;
 };
 
-export const TransactionList = ({ transactions }: AccountTransactionProps) => {
+export const TransactionList = ({
+    transactions,
+    accountName,
+    account,
+    page,
+    setPage,
+}: AccountTransactionProps) => {
     const { utils } = useNativeStyles();
     const accountTransactionsByDate = groupTransactionsByDate(transactions);
-    const transactionDateKeys = Object.keys(accountTransactionsByDate);
+    const transactionDateKeys = D.keys(accountTransactionsByDate);
+    const totalAccountsTransactions = useSelector((state: AccountsRootState) =>
+        selectAccountsTransactionsCount(state, account.key),
+    );
+    const flatListRef = useRef<FlatList | null>(null);
 
     if (A.isEmpty(transactionDateKeys))
         // TODO Temporary loading state just so it's visible that transactions are loading
         return <ActivityIndicator size="large" color={utils.colors.forest} />;
 
+    const handleSetPage = (page: number) => {
+        setPage(page);
+        flatListRef?.current?.scrollToOffset({
+            animated: true,
+            offset: 0,
+        });
+    };
+
     return (
-        <>
-            {transactionDateKeys.map(dateKey => (
-                <TransactionListGroup dateKey={dateKey} key={dateKey}>
-                    {accountTransactionsByDate[dateKey].map(transaction => (
+        <FlatList
+            ref={flatListRef}
+            contentContainerStyle={{ paddingBottom: TAB_BAR_HEIGHT }}
+            ListHeaderComponent={
```

Comment:
nit: what about having a `components` prop? It's a more common pattern IMO

Reply:
@dahaca Maybe I dont understand your nitpick correctly but this component is React native cmp and the API is already predetermined as you can see here. https://reactnative.dev/docs/sectionlist#listheadercomponent

If your comment was about sth else, could you please elaborate?

Reply:
Damn, I'm sorry! Assumed it was ours...